src/org/zaproxy/zap/extension/callback/ui/CallbackPanel.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2017 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.callback.ui;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.AbstractPanel;
+import org.zaproxy.zap.extension.callback.ExtensionCallback;
+import org.zaproxy.zap.utils.DisplayUtils;
+import org.zaproxy.zap.utils.FontUtils;
+
+import javax.swing.Box;
+import javax.swing.ImageIcon;
+import javax.swing.JButton;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JToolBar;
+import java.awt.CardLayout;
+import java.awt.GridBagConstraints;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+public class CallbackPanel extends AbstractPanel {
+
+    private static final long serialVersionUID = 1L;
+
+    private JToolBar mainToolBar;
+    private JPanel mainPanel = null;
+    private JScrollPane scrollPane = null;
+    private CallbackTable resultsTable = null;
+    private CallbackTableModel model = null;
+    private ExtensionCallback extensionCallback;
+
+    public CallbackPanel(ExtensionCallback extensionCallback) {
+        super();
+        this.extensionCallback = extensionCallback;
+        initialize();
+    }
+
+    private void initialize() {
+        this.setLayout(new CardLayout());
+        this.setName(Constant.messages.getString("callback.panel.name"));
+        this.setIcon(new ImageIcon(CallbackPanel.class.getResource("/resource/icon/16/callbacks.png")));
+        this.add(getMainPanel(), getMainPanel().getName());
+    }
+
+    private javax.swing.JPanel getMainPanel() {
+        if (mainPanel == null) {
+            mainPanel = new JPanel(new java.awt.GridBagLayout());
```

Comment:
Lets use `BorderLayout` instead, it does not require extra configurations (`GridBagConstraints`).